ansible_base/rbac/evaluations.py

```diff
@@ -0,0 +1,115 @@
+from django.conf import settings
+
+from ansible_base.rbac import permission_registry
+from ansible_base.rbac.models import ObjectRole, get_evaluation_model
+
+'''
+RoleEvaluation or RoleEvaluationUUID models are the authority for permission evaluations,
+meaning, determining whether a user has a permission to an object.
+
+Methods needed for producing querysets (of objects a user has a permission to
+or users that have a permission to an object) or making single evaluations
+are defined on the RoleEvaluation model.
+
+This module has logic to attach those evaluation methods to the external
+models in an app using these RBAC internals.
+'''
+
+
+def validate_codename_for_model(codename, model):
+    if '_' not in codename:
+        # convience to call JobTemplate.accessible_objects(u, 'execute')
+        name = f'{codename}_{model._meta.model_name}'
+    else:
+        # sometimes permissions are referred to with the app name, like test_app.say_cow
+        if '.' in codename:
+            name = codename.split('.')[-1]
+        else:
+            name = codename
+    if name.startswith('add'):
+        if model._meta.model_name != 'organization':
```

Comment:
This is untenable for hub. A lot of our "add" actions are based on some other parent objects. For example when a user adds a collection to the system, we check to see if they have "add" permissions on the namespace. This can't just be hardcoded to only work with "organization".

We also want to support global add permissions.

Reply:
This was sloppy, the "organization" string shouldn't be hard coded anywhere like this. The criteria at this line should be if the model has no child models.